google/cloud/bigtable/batcher.py

```diff
@@ -51,36 +53,54 @@ def __init__(self, max_mutation_bytes=MAX_MUTATION_SIZE, flush_count=FLUSH_COUNT
         self.total_size = 0
         self.max_mutation_bytes = max_mutation_bytes
         self.flush_count = flush_count
+        self._lock = threading.Lock()
 
     def get(self):
-        """Retrieve an item from the queue. Recalculate queue size."""
-        row = self._queue.get()
-        mutation_size = row.get_mutations_size()
-        self.total_mutation_count -= len(row._get_mutations())
-        self.total_size -= mutation_size
-        return row
+        """
+        Retrieve an item from the queue. Recalculate queue size.
+
+        If the queue is empty, return None.
+        """
+        with self._lock:
+            try:
+                row = self._queue.get_nowait()
```

Comment:
I think we only need to lock on the queue, can the rest of the logic be moved out of the lock?